Mapping/AttributeMetadata.php

```diff
@@ -97,6 +105,7 @@ class AttributeMetadata implements AttributeMetadataInterface
     public function __construct($name)
     {
         $this->name = $name;
+        $this->isIdentifier = false;
```

Comment:
Can't it be part of the constructor (with a default value) ? Even if it's not the case right now, I'd expect this attribute metadata to be immutable as much as possible.